diff --git a/dwm.c b/dwm.c
index a820b57..95b18ad 100644
--- a/dwm.c
+++ b/dwm.c
@@ -913,6 +913,8 @@ focus(Client *c)
 		XDeleteProperty(dpy, root, netatom[NetActiveWindow]);
 	}
 	selmon->sel = c;
+	if (selmon->sel)
+		resizeclient(c, c->x, c->y, c->w, c->h);
 	drawbars();
 }
 
@@ -1405,7 +1407,13 @@ resizeclient(Client *c, int x, int y, int w, int h)
 	c->oldy = c->y; c->y = wc.y = y;
 	c->oldw = c->w; c->w = wc.width = w;
 	c->oldh = c->h; c->h = wc.height = h;
-	wc.border_width = c->bw;
+	if (c == selmon->sel)
+		wc.border_width = c->bw;
+	else {
+		wc.border_width = 0;
+		wc.x += c->bw;
+		wc.y += c->bw;
+	}
 	if (((nexttiled(c->mon->clients) == c && !nexttiled(c->next))
 	    || &monocle == c->mon->lt[c->mon->sellt]->arrange)
 	    && !c->isfullscreen && !c->isfloating
@@ -1908,6 +1916,9 @@ unfocus(Client *c, int setfocus)
 		XSetInputFocus(dpy, root, RevertToPointerRoot, CurrentTime);
 		XDeleteProperty(dpy, root, netatom[NetActiveWindow]);
 	}
+	if (c == selmon->sel)
+		selmon->sel = NULL;
+	resizeclient(c, c->x, c->y, c->w, c->h);
 }
 
 void
