--- b/dwm.c	2019-09-18 20:11:54.060598921 +0200
+++ a/dwm.c	2019-09-18 21:00:54.844923573 +0200
@@ -1216,6 +1205,12 @@
 	c->bw = borderpx;
 
 	wc.border_width = c->bw;
+
+	if (c->x == selmon->wx)
+	  c->x += (c->mon->ww - WIDTH(c)) / 2 - c->bw;
+	if (c->y == selmon->wy)
+	  c->y += (c->mon->wh - HEIGHT(c)) / 2 - c->bw;
+
 	XConfigureWindow(dpy, w, CWBorderWidth, &wc);
 	XSetWindowBorder(dpy, w, scheme[SchemeNorm][ColBorder].pixel);
 	configure(c); /* propagates border_width, if size doesn't change */
